Week-6/Assignment_6.1_MaxSum.js

```diff
@@ -0,0 +1,36 @@
+const maxSubArray = (nums) => {
+    // initiate two variable, maxSum for total max, sum for current max
+    let maxSum = -Infinity
+    let currentSum = 0
+    // iterate through the nums, store sub-problems result
+    for(let i = 0; i < nums.length; i++){ 
+        //cumulating answers to the top
+        const maxSubArray = (nums) => {
+    // initiate two variable, maxSum for total max, sum for current max
+    let maxSum = -Infinity
+    let currentSum = 0
+    // iterate through the nums, store sub-problems result
+    for(let i = 0; i < nums.length; i++){ 
+        //cumulating answers to the top
+        
+        //compare currentSum add current number 
+        //with current number and store the maximum value
+        currentSum = Math.max(nums[i], currentSum + nums[i])
+        
+        //compare maxSum with currentSum and store the greater value
+        maxSum = Math.max(currentSum, maxSum)
+        
+    }
+    return maxSum
+        //compare currentSum add current number 
+        //with current number and store the maximum value
+        currentSum = Math.max(nums[i], currentSum + nums[i])
+        
+        //compare maxSum with currentSum and store the greater value
+        maxSum = Math.max(currentSum, maxSum)
+        
+    }
+
+    return maxSum;

```

Comment:
Good work.

1. Mention TC & SC of the solution
2. Handle edge cases like when array is null or undefined your code will throw error

Reply:
3. Try for more optimized solution